include/deal.II/base/exceptions.h

```diff
@@ -1769,6 +1769,35 @@ namespace internal
       }
 #  endif
 
+/**
+ * An assertion that checks that the kernel was launched and executed
+ * successfully.
+ *
+ * @note This and similar macro names are examples of preprocessor definitions
+ * in the deal.II library that are not prefixed by a string that likely makes
+ * them unique to deal.II. As a consequence, it is possible that other
+ * libraries your code interfaces with define the same name, and the result
+ * will be name collisions (see
+ * https://en.wikipedia.org/wiki/Name_collision). One can <code>\#undef</code>
+ * this macro, as well as all other macros defined by deal.II that are not
+ * prefixed with either <code>DEAL</code> or <code>deal</code>, by including
+ * the header <code>deal.II/base/undefine_macros.h</code> after all other
+ * deal.II headers have been included.
+ *
+ * @ingroup Exceptions
+ * @author Bruno Turcksin, 2020
+ */
+#  ifdef DEBUG
+#    define AssertCudaKernel                          \
+      cudaError_t error_code = cudaPeekAtLastError(); \
+      AssertCuda(error_code);                         \
```

Comment:
Any reason to not just do
```suggestion
      AssertCuda(cudaPeekAtLastError());                         \
```
as we did in most places before?

Reply:
You'll have to wrap this piece of code in `{...}`. That's because you're creating a local variable, and if you don't create a scope for it, you can't have two `AssertCudaKernel` statements in the same scope.

Reply:
@masterleinad since I need to create an `error_code` if I don't want to call `cudaDeviceSynchronize` twice, I decided to use the `error_code` earlier so that I do not call `cudaPeekAtLastError()` twice
@bangerth done